Frameworks/UIKit/_UIGestureCoordinator.mm

```diff
@@ -0,0 +1,445 @@
+    // rebuild tracking gesture list if this flag is YES. otherwise, don't re-build the tracking gestures
+    BOOL _resetAllTrackingGestures;
+
+    // gesture recognizer tracking list. Prefer NSMutableSortedSet, but we don't support yet.
+    // using NSMutableArray instead, same reason for _dManipRecognizers
+    // it gets rebuild when processing a touch if _resetAllTrackingGestures 
+    NSMutableArray* _gestureTrackingList;
+
+    // system DManip gesture recognizer list
+    // when _resetAllTrackingGestures is set to YES, during touch processing
+    // we will rebuild DManip gesture list
+    NSMutableArray* _dManipRecognizers;
+
+    // gesture list used for book keeping gestures to be removed in one Touch
+    // processing cycle. When processing a Touch event, some gestures can be
+    // cancelled or be recognized (ended), thus become ended. These gestures will be
+    // stored in this variable and then cleaned up. So when processing next touch
+    // we don't have to enumerate through these ended gestures, improve enumeration efficiency
+    NSMutableSet* _gesturesToRemove;
+
+    // gesture list used for book keeping gestures in UIGestureRecognizerStatePossible.  
+    // when one candidate gesture in _gestureTrackingList becomes active we need enumerate 
+    // this list to make sure that if all gestures in this list can be fired concurrently, 
+    // if not, they will be transit to cancelled state and removed from this list.
+    // it improves gesture processing efficiency
+    NSMutableSet* _gesturesInPossibleState;
+}
+
++ (instancetype)singleton {
+    if (!g_UIGestureCoordinator) {
+        g_UIGestureCoordinator = [_UIGestureCoordinator new];
+    }
+
+    return g_UIGestureCoordinator;
+}
+
+- (instancetype)init {
+    if (self = [super init]) {
+        _resetAllTrackingGestures = YES;
+        _gesturesToRemove = [NSMutableSet new];
+        _gesturesInPossibleState = [NSMutableSet new];
+        _gestureTrackingList = [NSMutableArray new];
+        _dManipRecognizers = [NSMutableArray new];
+    }
+
+    return self;
+}
+
+// test override to provide custom gestures
+- (void)buildTrackingGesturesListWith:(NSMutableArray*)regularRecognizers
+                     PossibleGestures:(NSMutableSet*)gesturesInPossibleState
+                        DManipGesture:(NSMutableArray*)dManipRecognizers {
+    // by default do nothing
+}
+
+// build tracking gesture list given a view, if view is nil, it invokes buildTrackingGesturesListWith override to get
+// custom build list of gesture for testing
+- (void)_buildTrackingGesturesListForView:(UIView*)view {
+    if (view == nil) {
+        if (_resetAllTrackingGestures) {
+            _resetAllTrackingGestures = NO;
+            return [self buildTrackingGesturesListWith:_gestureTrackingList
+                                      PossibleGestures:_gesturesInPossibleState
+                                         DManipGesture:_dManipRecognizers];
+        }
+    }
```

Comment:
it would remove the fork here and test the below production code...

---
In reply to: [100647623](https://github.com/Microsoft/WinObjC/pull/1909#discussion_r100647623) [](ancestors = 100647623)